archspec/cpu/detect.py

```diff
@@ -206,11 +206,26 @@ def host():
     # Get a list of possible candidates for this micro-architecture
     candidates = compatible_microarchitectures(info)
 
+    # Sorting criteria for candidates
+    def sorting_fn(item):
+        return len(item.ancestors), len(item.features)
+
+    # Get the best generic micro-architecture
+    generic_candidates = [c for c in candidates if c.vendor == "generic"]
+    best_generic = sorted(generic_candidates, key=sorting_fn, reverse=True)[0]
```

Comment:
Why not use `min` or `max` if you're just going to take the 0th element?

Reply:
Done. I could save a couple of lines if we were supporting only Python 3.X:
```python
return max(candidates, key=sorting_fn, default=best_generic)
```
Unfortunately Python 2.X doesn't have a `default` keyword arg for `max`.